File: src/main/java/com/example/desafio/api/models/entitys/BrandEntity.java
package com.example.desafio.api.models.entitys;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name = "brands")
public class BrandEntity extends BaseEntity {

    @Column(name = "name_brand")
    private String nameBrand;

    @OneToMany(mappedBy = "brandId", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ModelCarEntity> modelCars;

    public BrandEntity(String nameBrand, List<ModelCarEntity> modelCars) {
        this.nameBrand = nameBrand;
        this.modelCars = modelCars;
    }

    public BrandEntity() {
    }
}

File: src/main/kotlin/com/example/desafio/api/models/entitys/BrandEntity.kt
package com.example.desafio.api.models.entitys

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank


@Entity
@Table(name = "brands")
data class BrandEntity(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long = 0L,

        var deleted: Boolean = false,

        @Column(name = "name_brand")
        var nameBrand: String? = null,

        @OneToMany(mappedBy = "brandId", cascade = [CascadeType.ALL], orphanRemoval = true)
        var modelCars: List<ModelCarEntity>? = null
)